rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/SecureCxfClientFactoryBuilder.java

```diff
@@ -0,0 +1,351 @@
+            if (serviceReference != null) {
+                STSClientConfiguration service = bundleContext.getService(serviceReference);
+                if (service != null) {
+                    return service;
+                }
+            }
+        }
+        throw new SecurityServiceException("Could not get STSClientConfiguration");
+    }
+
+    /**
+     * @see SecureCxfClientFactoryBuilder#buildFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass)
+            throws SecurityServiceException {
+        return buildFactory(endpointUrl, interfaceClass, null, null);
+    }
+
+    /**
+     * @see SecureCxfClientFactoryBuilder#buildFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass,
+            String username, String password) throws SecurityServiceException {
+        return buildFactory(endpointUrl, interfaceClass, username, password, null, false);
+    }
+
+    /**
+     * Creates a factory that will return security-aware clients.
+     *
+     * @see SecureCxfClientFactoryBuilder.SecureCxfClientFactory
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass,
+            String username, String password, List<?> providers, boolean disableCnCheck)
+            throws SecurityServiceException {
+        if (StringUtils.isEmpty(endpointUrl) || interfaceClass == null) {
+            throw new IllegalArgumentException(
+                    "Called without a valid URL, will not be able to connect.");
+        }
+
+        JAXRSClientFactoryBean jaxrsClientFactoryBean = new JAXRSClientFactoryBean();
+        jaxrsClientFactoryBean.setServiceClass(interfaceClass);
+        jaxrsClientFactoryBean.setAddress(endpointUrl);
+        jaxrsClientFactoryBean.setClassLoader(interfaceClass.getClassLoader());
+        jaxrsClientFactoryBean.getInInterceptors().add(new LoggingInInterceptor());
+        jaxrsClientFactoryBean.getOutInterceptors().add(new LoggingOutInterceptor());
+
+        if (!CollectionUtils.isEmpty(providers)) {
+            jaxrsClientFactoryBean.setProviders(providers);
+        }
+
+        Client cxfClient = WebClient.client(jaxrsClientFactoryBean.create(interfaceClass));
+        if (cxfClient == null) {
+            throw new SecurityServiceException("Could not construct base client");
+        }
+        ClientConfiguration clientConfig = WebClient.getConfig(cxfClient);
+
+        if (!StringUtils.startsWithIgnoreCase(endpointUrl, "https")) {
+            LOGGER.warn("Cannot secure non-https connection " + endpointUrl
+                    + ", only unsecured clients will be created");
+        } else {
+            initSecurity(clientConfig, cxfClient, username, password);
+        }
+
+        if (disableCnCheck) {
```

Comment:
Move this disableCnCheck into the else above (right after initSecurity). It doesn't make sense to call this if we are http only. (Somebody might call the buildFactory with the flag set true).